examples/benchmarks/qaoa.py

```diff
@@ -10,20 +10,20 @@
 parser = argparse.ArgumentParser()
 parser.add_argument("--nqubits", default=4, help="Number of qubits.", type=int)
 parser.add_argument("--nangles", default=4, help="Number of layers.", type=int)
-parser.add_argument("--trotter", action="store_true", help="Use TrotterHamiltonian.")
+parser.add_argument("--dense", action="store_true", help="Use dense Hamiltonian or terms.")
 parser.add_argument("--solver", default="exp", help="Solver to apply gates.", type=str)
 parser.add_argument("--method", default="Powell", help="Optimization method.", type=str)
 parser.add_argument("--maxiter", default=None, help="Maximum optimization iterations.", type=int)
 
 
-def main(nqubits, nangles, trotter=False, solver="exp",
+def main(nqubits, nangles, dense=True, solver="exp",
          method="Powell", maxiter=None):
     """Performs a QAOA minimization test."""
 
     print("Number of qubits:", nqubits)
     print("Number of angles:", nangles)
 
-    hamiltonian = hamiltonians.XXZ(nqubits=nqubits, trotter=trotter)
+    hamiltonian = hamiltonians.XXZ(nqubits=nqubits, dense=dense)
```

Comment:
I am trying to read the docs from the user perspective and the usage of the dense flag and trotterization, e.g. what happens if we sent dense to True or False and trotterization, etc. We should include a comment about that in the advanced examples "Using Trotter decomposition", explaining the advantage of dense=False and the interoperability with dense=True.

Reply:
We already had a note regarding the `dense` flag in the Hamiltonians section, so I extended this and I added a reference to this in the "Using Trotter decomposition example". Please have a look and let me know if you agree.